src/rpc/core_rpc_server.cpp

```diff
@@ -616,6 +616,23 @@ namespace cryptonote
       return true;
     }
 
+    unsigned int concurrency_count = std::thread::hardware_concurrency() * 4;
```

Comment:
Why is this multiplied by 4? And unfortunately we have been avoiding `std::thread` because mingw32 and friends do not support it. We have been using `boost::thread` as a stop-gap.

There is also a function in `src/common/util.h` called `get_max_concurrency`. That function returns the lesser of `boost::thread::hardware_concurrency` and the `--max-concurrency` argument. @moneromooo-monero @hyc @fluffypony which should take precedent here - command line `--max-concurrency` or the mining threads RPC? Lesser of the two?